spec/ui/editor_spec.moon

```diff
@@ -281,6 +281,25 @@ describe 'Editor', ->
       editor.buffer = buffer2
       assert.equal 15, cursor.pos
 
+    it 'updates .last_shown for buffer switched out', ->
+      time = os.time
+      now = time!
+      os.time = -> now
+      pcall ->
+        editor.buffer = Buffer {}
+      os.time = time
+      assert.same now, buffer.last_shown
+
+  context 'previewing', ->
+    it 'does not update last_shown for previewed or switched out buffer', ->
```

Comment:
Yes, that makes sense. This isn't currently a problem with the interaction, since that always restores the original buffer, but I'm thinking that there is a general case where a preview is shown in an editor and the original buffer isn't restored. In that case we fail to properly update `last_shown` for the switched out buffer.